prow/config/inrepoconfig.go

```diff
@@ -0,0 +1,126 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/git"
+	"k8s.io/test-infra/prow/github"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	inRepoConfigFileName = ".prow.yaml"
+)
+
+// ProwYAML represents the content of a .prow.yaml file
+// used to version Presubmits inside the tested repo.
+type ProwYAML struct {
+	Presubmits []Presubmit `json:"presubmits"`
+}
+
+// ProwYAMLGetter is used to retrieve a ProwYAML. Tests should provide
+// their own implementation and set that on the Config.
+type ProwYAMLGetter func(c *Config, gc *git.Client, identifier, baseSHA string, headSHAs ...string) (*ProwYAML, error)
+
+// Verify defaultProwYAMLGetter is a ProwYAMLGetter
+var _ ProwYAMLGetter = defaultProwYAMLGetter
+
+func defaultProwYAMLGetter(
+	c *Config,
+	gc *git.Client,
+	identifier string,
+	baseSHA string,
+	headSHAs ...string) (*ProwYAML, error) {
+
+	log := logrus.WithField("repo", identifier)
+	log.Debugf("Attempting to get %q.", inRepoConfigFileName)
+
+	if gc == nil {
+		log.Error("defaultProwYAMLGetter was called with a nil git client")
+		return nil, errors.New("gitClient is nil")
+	}
+
+	repo, err := gc.Clone(identifier)
+	if err != nil {
+		return nil, fmt.Errorf("failed to clone repo for %q: %v", identifier, err)
+	}
+	defer func() {
+		if err := repo.Clean(); err != nil {
+			log.WithError(err).Error("Failed to clean up repo.")
+		}
+	}()
+
+	if err := repo.Config("user.name", "prow"); err != nil {
+		return nil, err
+	}
+	if err := repo.Config("user.email", "prow@localhost"); err != nil {
+		return nil, err
+	}
+	if err := repo.Config("commit.gpgsign", "false"); err != nil {
+		return nil, err
+	}
+
+	var mergeMethod github.PullRequestMergeType
+	identifierSlashSplit := strings.Split(identifier, "/")
+	if len(identifierSlashSplit) == 2 {
+		mergeMethod = c.Tide.MergeMethod(identifierSlashSplit[0], identifierSlashSplit[1])
+		log.Debugf("Using merge strategy %q.", mergeMethod)
+	} else {
+		log.Debugf("Defaulting merge strategy to %q.", github.MergeMerge)
```

Comment:
Why doesn't `tide.MergeMethod()` do this defaulting for us? Is this use-case here special? Is that the default for `tide` if it is not set, as well?

Reply:
Updated